interface/main/calendar/add_edit_event.php

```diff
@@ -2046,10 +2048,24 @@ function validateform(event,valu){
 
     //add rule if choose repeating event
     if ($('#form_repeat').is(':checked') || $('#days_every_week').is(':checked')){
+
+<?php switch ($GLOBALS['date_display_format']) {
+            case 0:
+                echo "var format = 'YYYY-MM-DD';";
+                break;
+            case 1:
+                echo "var format = 'MM/DD/YYYY';";
+                break;
+            case 2:
+                echo "var format = 'DD/MM/YYYY';";
+                break;
+} ?>
```

Comment:
Is there a way to avoid needing to add above to all scripts that will need to do this?

Reply:
We can add new function like getFormatDate_js().
But where.. maybe in the file - openemr/library/formatting_DateToYYYYMMDD_js.js.php ?

Reply:
I think that makes sense.

Reply:
Better solution -  combining the validateJS format with [DateFormatRead()](https://github.com/openemr/openemr/blob/master/library/formatting.inc.php#L147) (openemr/library/formatting.inc.php L147). 
Agree?

Reply:
Done

Reply:
agreed :)